src/Microsoft.AspNetCore.Mvc.RazorPages/Page.cs

```diff
@@ -139,6 +139,78 @@ protected RedirectResult Redirect(string url)
+
+        /// <summary>
+        /// Redirects (<see cref="StatusCodes.Status302Found"/>) to the specified <paramref name="page"/>.
+        /// </summary>
+        /// <param name="page">The name of the page.</param>
+        /// <param name="routeValues">The parameters for a route.</param>
+        /// <param name="fragment">The fragment to add to the URL.</param>
+        /// <returns>The <see cref="RedirectToPageResult"/>.</returns>
+        protected RedirectToPageResult RedirectToPage(string page, object routeValues, string fragment)
+             => new RedirectToPageResult(page, routeValues, fragment);
+
+        /// <summary>
+        /// Redirects (<see cref="StatusCodes.Status301MovedPermanently"/>) to the specified <paramref name="page"/>.
+        /// </summary>
+        /// <param name="page">The name of the page.</param>
+        /// <returns>The <see cref="RedirectToPageResult"/> with <see cref="RedirectToPageResult.Permanent"/> set.</returns>
+        protected RedirectToPageResult RedirectToPagePermanent(string page)
+             => RedirectToPagePermanent(page, routeValues: null);
+
+        /// <summary>
+        /// Redirects (<see cref="StatusCodes.Status301MovedPermanently"/>) to the specified <paramref name="page"/>.
+        /// </summary>
+        /// <param name="page">The name of the page.</param>
+        /// <param name="routeValues">The parameters for a route.</param>
+        /// <returns>The <see cref="RedirectToPageResult"/> with <see cref="RedirectToPageResult.Permanent"/> set.</returns>
+        protected RedirectToPageResult RedirectToPagePermanent(string page, object routeValues)
+            => RedirectToPagePermanent(page, routeValues, fragment: null);
+
+        /// <summary>
+        /// Redirects (<see cref="StatusCodes.Status301MovedPermanently"/>) to the specified <paramref name="page"/>.
+        /// </summary>
+        /// <param name="page">The name of the page.</param>
+        /// <param name="fragment">The fragment to add to the URL.</param>
+        /// <returns>The <see cref="RedirectToPageResult"/> with <see cref="RedirectToPageResult.Permanent"/> set.</returns>
+        protected RedirectToPageResult RedirectToPagePermanent(string page, string fragment)
+            => RedirectToPagePermanent(page, routeValues: null, fragment: fragment);
+
+        /// <summary>
+        /// Redirects (<see cref="StatusCodes.Status301MovedPermanently"/>) to the specified <paramref name="page"/>.
+        /// </summary>
+        /// <param name="page">The name of the page.</param>
+        /// <param name="routeValues">The parameters for a route.</param>
+        /// <param name="fragment">The fragment to add to the URL.</param>
+        /// <returns>The <see cref="RedirectToPageResult"/> with <see cref="RedirectToPageResult.Permanent"/> set.</returns>
+        protected RedirectToPageResult RedirectToPagePermanent(string page, object routeValues, string fragment)
```

Comment:
Do these mirror the set of methods we have on controller for `RedirectToAction`?